allennlp/models/classifier.py

```diff
@@ -0,0 +1,113 @@
+from typing import Any, Dict, List
+
+import torch
+from allennlp.data import Vocabulary
+from allennlp.models.model import Model
+from allennlp.modules import Seq2VecEncoder, TextFieldEmbedder
+from allennlp.nn import InitializerApplicator
+from allennlp.nn.util import get_text_field_mask
+from allennlp.training.metrics import CategoricalAccuracy
+
+
+@Model.register("classifier")
+class Classifier(Model):
+    """
+    This ``Model`` implements a generic text classifier. After embedding the text into
+    a text field, we will optionally encode the embeddings with a Seq2VecEncoder and
+    then pass the embeddings to a linear classification layer, which projects
+    into the label space.
+
+    Parameters
+    ----------
+    vocab : ``Vocabulary``
+    text_field_embedder : ``TextFieldEmbedder``
+        Used to embed the input text into a ``TextField``
+    encoder: ``Seq2VecEnoder``
+        Encoder layer for the input text
+    dropout : ``float``, optional (default=None)
+        Dropout percentage to use.
+    initializer : ``InitializerApplicator``, optional (default=``InitializerApplicator()``)
+        If provided, will be used to initialize the model parameters.
+    """
+    def __init__(self,
+                 vocab: Vocabulary,
+                 text_field_embedder: TextFieldEmbedder,
+                 encoder: Seq2VecEncoder = None,
+                 dropout: float = None,
+                 initializer: InitializerApplicator = InitializerApplicator()) -> None:
+
+        super().__init__(vocab)
+        self._text_field_embedder = text_field_embedder
+
+        if encoder:
+            self._encoder = encoder
+            self._classifier_input_dim = self._encoder.get_output_dim()
+        else:
+            self._encoder = None
+            self._classifier_input_dim = self._text_field_embedder.get_output_dim()
+
+        if dropout:
+            self._dropout = torch.nn.Dropout(dropout)
+        else:
+            self._dropout = None
+
+        self._num_labels = vocab.get_vocab_size(namespace="labels")
```

Comment:
I think this should be more configurable, definitely you should be able to specify a different namespace, possibly you should be allowed to set it explicitly as well?